tests/components/zeroconf/test_init.py

```diff
@@ -1030,3 +1030,23 @@ async def test_no_name(hass, mock_async_zeroconf):
     register_call = mock_async_zeroconf.async_register_service.mock_calls[-1]
     info = register_call.args[0]
     assert info.name == "Home._home-assistant._tcp.local."
+
+
+async def test_service_info_compatibility(hass):
+    """Test compatibility with old-style dict.
+
+    To be removed in 2022.6
+    """
+    discovery_info = zeroconf.ZeroconfServiceInfo(
+        host="mock_host",
+        port=None,
+        hostname="mock_hostname",
+        type="mock_type",
+        name="mock_name",
+        properties={},
+    )
+
+    assert discovery_info["host"] == "mock_host"
+    assert discovery_info.get("host") == "mock_host"
+    assert discovery_info.get("host", "fallback_host") == "mock_host"
+    assert discovery_info.get("invalid_key", "fallback_host") == "fallback_host"
```

Comment:
We assert if the expected error is logged.

Reply:
Once this one is agreed upon and merged, I will create a separate PR for the other four components (`dhcp`/`mqtt`/`usb`/`ssdp`)